polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.h

```diff
@@ -31,9 +31,9 @@ void populateBareMemRefToLLVMConversionPatterns(LLVMTypeConverter &converter,
 #define GEN_PASS_DECL
 #include "mlir/Dialect/Polygeist/Transforms/Passes.h.inc"
 
-std::unique_ptr<Pass> createParallelLICMPass();
 std::unique_ptr<Pass> createMem2RegPass();
 std::unique_ptr<Pass> createLICMPass();
+std::unique_ptr<Pass> createLICMPass(bool relaxedAliasing);
```

Comment:
strict aliasing is not directly related to LICM, it should impact AliasAnalysis generally. Could AliasAnalysis read this info somewhere else (like Options.h) instead of passing through LICM pass?

Reply:
I know. AliasAnalysis is not invoked from cgeist directly and also it cannot use the polygeist options header given it is in the SYCL-MLIR folder. Also -relaxed-aliasing is a cgeist option, so is up to cgeist to use that option and pass it to the optimization passes that need it.

Reply:
I like this option being passed as an argument to `LICM`. However, the argument here should not be `bool`, but a `LICMOptions` holding the `bool` (add `options` field to `LICM` pass in `.td` file (see `-convert-sycl-to-llvm` for reference). This will include updating how we define passes in `polygeist` (it uses the old way). I'll push a PR now so you can rebase and use the new way of defining passes and get the option the canonical way (you'll also be able to pass in the CL for testing).

Reply:
OK will rebase once https://github.com/intel/llvm/pull/8513 is merged.